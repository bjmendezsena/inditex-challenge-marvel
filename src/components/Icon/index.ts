export * from './FavIcon';
export * from './MagnifyingGlassIcon';
export * from './types';
