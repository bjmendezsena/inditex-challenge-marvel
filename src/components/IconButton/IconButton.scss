@import '@/styles/variables.scss';

.iconButton {
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: transparent;
  border: none;
  cursor: pointer;
  margin: 0;
  transition: all 0.3s ease;
  width: min-content;
  height: min-content;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 100%;
  padding: 10px;
  color: $secondary-color;
  &:hover {
    background-color: $muted-color;
  }
}
