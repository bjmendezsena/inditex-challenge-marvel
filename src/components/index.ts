export * from "./FavButton";
export * from "./Header";
export * from "./Icon";
export * from "./Icon";
export * from "./IconButton";
export * from "./Input";
export * from "./Spinner";
export * from "./AsyncImage";
