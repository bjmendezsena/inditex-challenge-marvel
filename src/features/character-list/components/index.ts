export * from './CharacterCard';
export * from './CharacterCardSkeleton';
export * from './CharacterList';
