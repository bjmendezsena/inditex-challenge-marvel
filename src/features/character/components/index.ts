export * from './CharacterHeaderPage';
export * from './ComicsSection';
