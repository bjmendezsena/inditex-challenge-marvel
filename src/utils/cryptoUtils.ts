import Crypto from "crypto-js";

export function createHashMd5(text: string): string {
  return Crypto.MD5(text).toString();
}
