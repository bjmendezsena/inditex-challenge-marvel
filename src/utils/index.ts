export * from "./cryptoUtils";
export * from "./urls";
export * from "./format";
