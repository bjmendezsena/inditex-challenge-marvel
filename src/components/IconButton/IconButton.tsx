import { ComponentProps } from "react";
import "./IconButton.scss";

export type IconButtonProps = {
  children: JSX.Element;
} & ComponentProps<"button">;
export const IconButton = ({ children, ...props }: IconButtonProps) => {
  return (
    <button {...props} className={"iconButton"}>
      {children}
    </button>
  );
};
