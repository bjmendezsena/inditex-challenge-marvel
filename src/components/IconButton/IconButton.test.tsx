import { render } from '@/tests/utils';
import { FavIcon } from '@/components';
import { IconButton } from './IconButton';

describe(`<${IconButton.name} />`, () => {
  it('should render', () => {
    const { container } = render(
      <IconButton>
        <FavIcon />
      </IconButton>
    );
    expect(container).toMatchSnapshot();
  });
});
