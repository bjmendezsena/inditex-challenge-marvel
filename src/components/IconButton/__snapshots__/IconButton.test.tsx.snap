// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<IconButton /> > should render 1`] = `
<div>
  <button
    class="iconButton"
  >
    <svg
      class="icon icon__size--md"
      fill="none"
      viewBox="0 0 56 38"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="m20 11.803-6-3.641-6 3.641v7.804l12 10.231 12-10.231v-7.803l-6-3.642-6 3.641Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;
